step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java

```diff
@@ -152,6 +158,7 @@ public void removeAutomationPackage(ObjectId id, ObjectPredicate objectPredicate
         AutomationPackage automationPackage = getAutomationPackageById(id, objectPredicate);
         deleteAutomationPackageEntities(automationPackage);
         automationPackageAccessor.remove(automationPackage.getId());
+        automationPackageLocks.removeLock(automationPackage.getId().toHexString());
```

Comment:
If the AP isn't removed we don't remove the lock. Not sure if this is an issue as we don't expect that much AP...

Reply:
We could try some cleanup mechanism, but each time an execution involving the AP is executed the lock would be recreated, not sure if it's worth it.